internal/vendors/accessor.go

```diff
@@ -35,101 +35,99 @@ func (p *postgresVendorAccessor) GetSomeStuff(ctx context.Context) ([]string, er
 	return results, nil
 }
 
-func (p *postgresVendorAccessor) GetAll(ctx context.Context, spec database.PaginationSpec) (*AccessorGetAllPaginationData, error) {
-	args := database.BuildPaginationArgs(spec)
-
-	dataQuery := `SELECT 
-		"id",
-		"name",
-		"description",
-		"bp_id",
-		"bp_name",
-		"rating",
-		"area_group_id",
-		"area_group_name",
-		"sap_code",
-		"modified_date",
-		"modified_by",
-		"dt" 
-		FROM vendor
-		ORDER BY created_at $1
-		LIMIT $2
-		OFFSET $3
-		`
-
-	rows, err := p.db.Query(dataQuery, args.Order, args.Limit, args.Offset)
-	if err != nil {
-		return nil, err
+func (p *postgresVendorAccessor) GetAll(ctx context.Context, spec GetAllVendorSpec) (*AccessorGetAllPaginationData, error) {
+	paginationArgs := database.BuildPaginationArgs(spec.PaginationSpec)
+
+	// Initialize clauses and arguments
+	var (
+		joinClauses     []string
+		whereClauses    []string
+		extraClauses    []string
+		args            []interface{}
+		i               = 1
```

Comment:
I think we could name this to something more verbose, like 
```suggestion
		argsIndex               = 1
```